ibm/resource_ibm_appid_token_config.go

```diff
@@ -0,0 +1,325 @@
+				Description: "A set of objects that are created when claims that are related to access tokens are mapped",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"source": {
+							Description:  "Defines the source of the claim. Options include: `saml`, `cloud_directory`, `facebook`, `google`, `appid_custom`, and `attributes`.",
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"saml", "cloud_directory", "appid_custom", "facebook", "google", "ibmid", "attributes", "roles"}, false),
+						},
+						"source_claim": {
+							Description: "Defines the claim as provided by the source. It can refer to the identity provider's user information or the user's App ID custom attributes.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"destination_claim": {
+							Description: "Optional: Defines the custom attribute that can override the current claim in token.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+					},
+				},
+			},
+			"id_token_claim": {
+				Description: "A set of objects that are created when claims that are related to identity tokens are mapped",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"source": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"saml", "cloud_directory", "appid_custom", "facebook", "google", "ibmid", "attributes", "roles"}, false),
+						},
+						"source_claim": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"destination_claim": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIBMAppIDTokenConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	appidClient, err := meta.(ClientSession).AppIDAPI()
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tenantID := d.Get("tenant_id").(string)
+
+	input := expandTokenConfig(d)
+
+	_, _, err = appidClient.PutTokensConfigWithContext(ctx, input)
+
+	if err != nil {
+		return diag.Errorf("Error updating AppID token configuration: %s", err)
+	}
+
+	d.SetId(tenantID)
+
+	return resourceIBMAppIDTokenConfigRead(ctx, d, meta)
+}
+
+func resourceIBMAppIDTokenConfigRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	appidClient, err := meta.(ClientSession).AppIDAPI()
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	tenantID := d.Id()
+
+	tokenConfig, _, err := appidClient.GetTokensConfigWithContext(ctx, &appid.GetTokensConfigOptions{
+		TenantID: &tenantID,
+	})
+
+	if err != nil {
+		return diag.Errorf("Error reading AppID token configuration: %s", err)
```

Comment:
Can we handle the situation when user deletes this configuration outside the TF using the cli or API...so on
then we may get 404 so if response is not nil and status code is 404 just set d.seID("") and retrun nil
```
if err != nil {
		if response != nil && response.StatusCode == 404 {
			d.SetId("")
			return nil
		}
		return diag.Errorf("Error reading AppID token configuration: %s", err)
	}
	```

Reply:
Theoretically it is not possible to delete `token configuration` although it is possible to delete whole AppID instance, in that case you would get 404, which I believe would be handled by another resource that creates AppID instance, ie instance would be re-created and thus tenantID would change. When tenantID changes all of AppID resources would force reload. I can still add this logic just in case.